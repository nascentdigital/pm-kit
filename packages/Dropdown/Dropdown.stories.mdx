import { Meta, Story, Preview, Props, Subtitle } from '@storybook/addon-docs/blocks'
import Dropdown from './Dropdown.jsx'
import {Default, LargeLabel, HideLabel, Error} from './Dropdown.stories.js'
import { version } from './package.json'

<Meta title="Design System|Dropdown" componenent={Dropdown} />

# Dropdown
<Subtitle>
version {version}
</Subtitle>

## Usage
* To install `Dropdown` package with npm run terminal:

```jsx
npm i @pm-kit/dropdown

```
* Import `Dropdown` in your componenent to render

```jsx
import Dropdown from '@pm-kit/dropdown'

```

## Properties 

<Props of={Dropdown} />

## Stories


#### Default
<Story name="Default">
 <Default />
</Story>

#### Large Label
<Story name="Large Label">
 <LargeLabel />
</Story>

#### Hide Label
<Story name="Hide Label">
 <HideLabel />
</Story>

#### Error
<Story name="Error">
 <Error />
</Story>

