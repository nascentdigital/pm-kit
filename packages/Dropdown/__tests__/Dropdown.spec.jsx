import React from 'react'
import { shallow } from 'enzyme'

import Dropdown from '../Dropdown'

describe('Dropdown', () => {
  it('renders default dropdown', () => {
    const dropdown = shallow(<Dropdown label="testing" onChange={() => {}} />)
    expect(dropdown).toMatchSnapshot()
  })
})
