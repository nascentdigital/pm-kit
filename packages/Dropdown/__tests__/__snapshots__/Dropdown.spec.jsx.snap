// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown renders default dropdown 1`] = `
<Dropdown
  forwardedRef={null}
  ignoreAccents={true}
  ignoreCase={true}
  label="testing"
  labelType="small"
  loadingState="success"
  matchFrom="start"
  onChange={[Function]}
  options={Array []}
  required={false}
  searchMaxLength={100}
  styles={Object {}}
  trim={true}
  type="text"
/>
`;
